### PR TITLE
Use generics on Reader

### PR DESCRIPTION
**Pull request by lcobucci**
Allowing for static analysis tools to more easily understand things.

### CODE REVIEW

File: lib/Doctrine/Common/Annotations/Reader.php [hunk 1]
```diff
@@ -26,9 +26,11 @@ public function getClassAnnotations(ReflectionClass $class);
      *
      * @param ReflectionClass $class          The ReflectionClass of the class from which
      *          the class annotations should be read.
-     * @param string          $annotationName The name of the annotation.
+     * @param class-string<T> $annotationName The name of the annotation.
      *
-     * @return object|null The Annotation or NULL, if the requested annotation does not exist.
+     * @return T|null The Annotation or NULL, if the requested annotation does not exist.
+     *
+     * @template T
```
Comment (lcobucci) [inline lib/Doctrine/Common/Annotations/Reader.php:33]: This is a very unusual place for `@template` :smile:

Comment (lcobucci) [reply lib/Doctrine/Common/Annotations/Reader.php:33]: Perhaps https://github.com/lcobucci/coding-standard/blob/7.0.x/src/Lcobucci/ruleset.xml#L17 can help?

**Comment by beberlei** [comment]
@lcobucci you can target this for 1.11 imho, its only changing comments. We did the same with Psalm improvements on ORM 2.7

File: lib/Doctrine/Common/Annotations/TokenParser.php [hunk 2]
```diff
@@ -12,6 +12,8 @@
 use const T_AS;
 use const T_COMMENT;
 use const T_DOC_COMMENT;
+use const T_NAME_FULLY_QUALIFIED;
```
Comment (lcobucci) [inline lib/Doctrine/Common/Annotations/TokenParser.php]: These issues were in the repo... but I guess CI didn't catch them before? Weeeird

Comment (lcobucci) [reply lib/Doctrine/Common/Annotations/TokenParser.php]: Not on 1.11.x, though :laughing:

**Comment by lcobucci** [comment]
> @lcobucci you can target this for 1.11 imho, its only changing comments. We did the same with Psalm improvements on ORM 2.7

@beberlei done!

**Comment by beberlei** [comment]
Yeah #372 has the same phpstan error, no clue how to work aorund that, i cant reproduce locally.

**Comment by lcobucci** [comment]
> Yeah #372 has the same phpstan error, no clue how to work aorund that, i cant reproduce locally.

@beberlei I was able to reproduce it locally... squeezed an extra commit in, explaining why it's okay to ignore these errors :+1:

**Comment by beberlei** [comment]
@lcobucci awesome, found the same workaround :D https://github.com/doctrine/annotations/pull/372/commits/2f69b8df153b4b60f9e580d002a53680c4d4576e

**Review by beberlei — APPROVED** [review]

**Pull request merged**